app/helpers.go

```diff
@@ -110,34 +110,20 @@ func NewSliceIterator(data []weave.Model) *SliceIterator {
 	}
 }
 
-// Valid implements Iterator and returns true iff it can be read
-func (s *SliceIterator) Valid() bool {
-	return s.idx < len(s.data)
-}
-
 // Next moves the iterator to the next sequential key in the database, as
 // defined by order of iteration.
 //
-// If Valid returns false, this method will panic.keys
-func (s *SliceIterator) Next() error {
+// Returns (nil, nil, errors.ErrDone) if there is no more data
+func (s *SliceIterator) Next() (key, value []byte, err error) {
 	if s.idx >= len(s.data) {
-		return errors.Wrap(errors.ErrDatabase, "Passed end of slice")
+		return nil, nil, errors.Wrap(errors.ErrDone, "slice iterator")
 	}
+	val := s.data[s.idx]
 	s.idx++
-	return nil
-}
-
-// Key returns the key of the cursor.
-func (s *SliceIterator) Key() (key []byte) {
-	return s.data[s.idx].Key
-}
-
-// Value returns the value of the cursor.
-func (s *SliceIterator) Value() (value []byte) {
-	return s.data[s.idx].Value
+	return val.Key, val.Value, nil
 }
 
-// Close releases the Iterator.
-func (s *SliceIterator) Close() {
+// Release releases the Iterator.
+func (s *SliceIterator) Release() {
```

Comment:
Note: this is a duplicate of `store.SliceIterator`. You could drop this implementation in favour of `store` without getting into cyclomatic dependencies.

Reply:
I did that before, but there were some complaints... I will put it back like that